kyo-data/shared/src/main/scala/kyo/Result.scala

```diff
@@ -220,24 +533,51 @@ object Result:
             catch
                 case ex => Panic(ex)
 
+        /** Converts the Result to an Either.
+          *
+          * @return
+          *   An Either with Left containing the error or exception, and Right containing the successful value
+          */
         def toEither: Either[E | Throwable, A] =
             fold(e => Left(e.getFailure))(Right(_))
 
+        /** Converts the Result to a Maybe.
+          *
+          * @return
+          *   A Maybe containing the successful value, or empty for non-Success Results
+          */
         def toMaybe: Maybe[A] =
             fold(_ => Maybe.empty)(Maybe(_))
 
+        /** Converts the Result to a Try.
+          *
+          * @param ev
+          *   Evidence that E is a subtype of Throwable
+          * @return
+          *   A Try containing the successful value, or Failure with the error
+          */
         def toTry(using
             @implicitNotFound("Fail type must be a 'Throwable' to invoke 'toTry'. Found: '${E}'")
             ev: E <:< Throwable
         ): Try[A] =
             fold(e => scala.util.Failure(e.getFailure.asInstanceOf[Throwable]))(scala.util.Success(_))
 
+        /** Swaps the success and failure cases of the Result.
+          *
+          * @return
+          *   A new Result with success and failure swapped
+          */
         def swap: Result[A, E] =
             self match
                 case Fail(e)    => Result.success(e)
                 case Success(v) => Result.fail(v)
                 case _          => self.asInstanceOf[Result[A, E]]
```

Comment:
Should we have a toString impl of `SuccessError`? `"Success(" * (depth - 1) + ...`

Reply:
Makes sense, I've added a `toString` to `DefinedEmpty` in `Maybe` as well: https://github.com/getkyo/kyo/pull/639/commits/852c90755c1761588cb88c235477e50d18cd3623